### PR TITLE
bada-bib: 0.4.1 -> 0.5.1

### PR DESCRIPTION
**Pull request by Cogitri**
###### Motivation for this change
New release that works w/ GTK4

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/bada-bib/default.nix [hunk 1]
```diff
@@ -9,23 +9,23 @@
 , glib
 , gnome
 , gobject-introspection
-, gtk3
+, gtk4
 , libxml2
 , pkg-config
 , python3Packages
 , wrapGAppsHook }:
```
Comment (bobby285271) [inline pkgs/applications/science/misc/bada-bib/default.nix:16]: We really want to use `wrapGAppsHook4` for all GTK4 apps :crying_cat_face:
Submitted https://github.com/RogerCrocker/BadaBib/pull/3

Comment (Cogitri) [reply pkgs/applications/science/misc/bada-bib/default.nix:16]: Ah thanks, missed that :)

Comment (bobby285271) [reply pkgs/applications/science/misc/bada-bib/default.nix:16]: The patch has been accepted, do you mind fetchpatch the patch and use `wrapGAppsHook4`?

Comment (RogerCrocker) [reply pkgs/applications/science/misc/bada-bib/default.nix:16]: Hi, I thought this issues might affect others as well, so I just made a new release with your patch included.

Comment (Cogitri) [reply pkgs/applications/science/misc/bada-bib/default.nix:16]: Thanks :smile:

**Pull request merged**